### PR TITLE
Stop using deprecated Fleet APIs

### PR DESCRIPTION
**Pull request by jsoriano**
Stop using Fleet APIs deprecated since 8.0 (https://www.elastic.co/guide/en/kibana/master/release-notes-8.0.0-rc1.html#deprecations-8.0.0-rc1).

The only one that seems we are using is:
>  Use `/epm/packages/{packageName}/{version}` instead of `/epm/packages/{pkgkey}`.

Kibana client retrieves its version from the status API on initialization.

Fixes https://github.com/elastic/elastic-package/issues/653

### CODE REVIEW

**Review by juliaElastic — APPROVED** [review]
LGTM

File: internal/kibana/test/httptest.go [hunk 1]
```diff
@@ -54,11 +57,11 @@ func pathForURL(url string) string {
 }
 
 func recordRequest(t *testing.T, r *http.Request, path string) {
-	client, err := NewClient()
+	client, err := stack.NewKibanaClient()
```
Comment (mrodm) [inline internal/kibana/test/httptest.go:60]: Good catch!

File: internal/kibana/test/httptest.go [hunk 2]
```diff
@@ -54,11 +57,11 @@ func pathForURL(url string) string {
...
 	require.NoError(t, err)
 
 	t.Logf("Recording %s in %s", r.URL.RequestURI(), path)
-	status, respBody, err := client.get(r.URL.RequestURI())
+	status, respBody, err := client.SendRequest(http.MethodGet, r.URL.RequestURI(), nil)
```
Comment (mrodm) [inline internal/kibana/test/httptest.go:64]: Why not doing public get, post, put and delete methods?

**Review by mrodm — APPROVED** [review]
:+1:

File: internal/kibana/test/httptest.go (hunk 2 above)
Comment (jsoriano) [reply internal/kibana/test/httptest.go:64]: I don't have a strong preference, I found better to expose the most generic one. Would you prefer to expose the ones per method?

**Pull request merged**